src/gui/gt_icons.cpp

```diff
@@ -17,29 +17,42 @@
 #include <QPainter>
 #include <QScreen>
 #include <QApplication>
-#include <QDesktopWidget>
+
+namespace
+{
+
+bool initialized = [](){
+    qRegisterMetaType<gt::gui::Icon>("GtIcon");
```

Comment:
Shouldn't this be adapted as well?
```suggestion
    qRegisterMetaType<gt::gui::Icon>("gt::gui::Icon");
```